pkg/cmd/cli/cmd/set/triggers.go

```diff
@@ -0,0 +1,744 @@
+			}
+			for _, s := range triggers.WebHooks {
+				fmt.Fprintf(w, "%s/%s\t%s\t%s\t%s\n", info.Mapping.Resource, info.Name, "webhook", s, "")
+			}
+			for _, s := range triggers.GitHubWebHooks {
+				fmt.Fprintf(w, "%s/%s\t%s\t%s\t%s\n", info.Mapping.Resource, info.Name, "github", s, "")
+			}
+			return nil
+		})
+		if err != nil {
+			fmt.Fprintf(w, "%s/%s\t%s\t%s\t%t\n", info.Mapping.Resource, info.Name, "<error>", "", false)
+		}
+	}
+	return nil
+}
+
+// updateTriggers updates only those fields with flags set by the user
+func (o *TriggersOptions) updateTriggers(triggers *TriggerDefinition) {
+	// clear everything
+	if o.RemoveAll {
+		*triggers = TriggerDefinition{}
+		return
+	}
+
+	// clear a specific field
+	if o.Remove {
+		if o.FromConfig {
+			triggers.ConfigChange = false
+		}
+		if len(o.FromImage) > 0 {
+			var newTriggers []ImageChangeTrigger
+			for _, trigger := range triggers.ImageChange {
+				if trigger.From != o.FromImage {
+					newTriggers = append(newTriggers, trigger)
+				}
+			}
+			triggers.ImageChange = newTriggers
+		}
+		if o.FromWebHook != nil {
+			triggers.WebHooks = nil
+		}
+		if o.FromGitHub != nil {
+			triggers.GitHubWebHooks = nil
+		}
+		return
+	}
+
+	// change the automated status
+	if o.Reset {
+		triggers.ConfigChange = o.Auto
+		for i := range triggers.ImageChange {
+			triggers.ImageChange[i].Auto = o.Auto
+		}
+		return
+	}
+
+	// change individual elements
+	if o.FromConfig {
+		triggers.ConfigChange = true
+	}
+	if len(o.FromImage) > 0 {
+		names := strings.Split(o.ContainerNames, ",")
+		if len(o.ContainerNames) == 0 {
+			names = nil
+		}
+		found := false
+		for i, trigger := range triggers.ImageChange {
+			if trigger.From == o.FromImage && trigger.Namespace == o.FromImageNamespace {
+				found = true
+				triggers.ImageChange[i].Auto = !o.Manual
+				triggers.ImageChange[i].Names = names
+				break
+			}
+		}
+		if !found {
+			triggers.ImageChange = append(triggers.ImageChange, ImageChangeTrigger{
+				From:      o.FromImage,
+				Namespace: o.FromImageNamespace,
+				Auto:      !o.Manual,
+				Names:     names,
+			})
+		}
+	}
+	if o.FromWebHook != nil {
+		triggers.WebHooks = []string{app.GenerateSecret(20)}
+	}
+	if o.FromGitHub != nil {
+		triggers.GitHubWebHooks = []string{app.GenerateSecret(20)}
+	}
+}
+
+type ImageChangeTrigger struct {
+	// If this trigger is automatically applied
+	Auto bool
+	// An ImageStreamTag name to target
+	From string
+	// The target namespace, normalized if set
+	Namespace string
+	// A list of names this trigger targets
+	Names []string
+}
+
+type TriggerDefinition struct {
+	ConfigChange   bool
+	ImageChange    []ImageChangeTrigger
+	WebHooks       []string
+	GitHubWebHooks []string
+}
+
+func defaultNamespace(namespace, defaultNamespace string) string {
+	if namespace == defaultNamespace {
+		return ""
+	}
+	return namespace
+}
+
+func NewDeploymentConfigTriggers(config *deployapi.DeploymentConfig) *TriggerDefinition {
```

Comment:
I would have called this Get and not New.  or NewTriggersFromDeploymentConfig
(same for NewBuildConfigTriggers below)